src/Simplex_tree/concept/FiltrationValue.h

```diff
@@ -5,19 +5,65 @@
-  * f(\sigma)\f$ whenever \f$\tau \subseteq \sigma\f$. Ordering the simplices 
-  * by increasing filtration values (breaking ties so as a simplex appears after 
-  * its subsimplices of same filtration value) provides an indexing scheme 
-  * (see IndexingTag).
-  */
-  struct FiltrationValue {
-    /** \brief Operator < is a StrictWeakOrdering. */
-    bool operator<(FiltrationValue f1, FiltrationValue f2);
-  };
+ *
+ * Needs to implement `std::numeric_limits<FiltrationValue>::has_infinity` and when it returns `true`,
+ * has to implement `std::numeric_limits<FiltrationValue>::infinity()`. If it returns `false`, has to
+ * implement `std::numeric_limits<FiltrationValue>::max()` instead.
+ *
+ * A <EM>filtration</EM> of a cell complex (see FilteredComplex) is
+ * a function \f$f:\mathbf{K} \rightarrow \mathbb{R}\f$ satisfying \f$f(\tau)\leq
+ * f(\sigma)\f$ whenever \f$\tau \subseteq \sigma\f$. Ordering the simplices
+ * by increasing filtration values (breaking ties so as a simplex appears after
+ * its subsimplices of same filtration value) provides an indexing scheme
+ * (see IndexingTag).
+ */
+struct FiltrationValue {
+  /**
+   * @brief Has to construct the default value of FiltrationValue. E.g., 0 for a numerical value or {} for a vector.
+   */
+  FiltrationValue(0);
+
+  // only for default ordering of filtration_vect_ in initialize_filtration
+  /**
+   * @brief Strictly smaller operator. If the filtration values are totally ordered, should be a StrictWeakOrdering.
+   */
+  bool operator<(FiltrationValue f1, FiltrationValue f2);
+  // only for prune_above_filtration
+  /**
+   * @brief Smaller or equal operator.
+   */
+  bool operator<=(FiltrationValue f1, FiltrationValue f2);
+  /**
+   * @brief Equality operator
+   */
+  bool operator==(FiltrationValue f1, FiltrationValue f2);
+  /**
+   * @brief Not equal operator
+   */
+  bool operator!=(FiltrationValue f1, FiltrationValue f2);
+
+  /**
+   * @brief Given two filtration values at which a simplex exists, returns the minimal union of births generating
+   * a lifetime including those two values. The overload for native arithmetic types like `double` or `int` is
+   * already implemented.
+   * If the filtration is 1-critical and totally ordered (as in one parameter persistence), the union is simply
+   * the minimum of the two values. If the filtration is 1-critical, but not totally ordered (possible for
+   * multi-persistence), than the union is also the minium if the two given values are comparable and the
+   * method should throw an error if they are not, as a same simplex should not exist at those two values.
+   * Finally, if the filtration is k-critical, FiltrationValue should be able to store an union of values and this
+   * method adds the values of @p f2 in @p f1 and removes the values from @p f1 which are comparable and greater than
+   * other values.
+   */
+  friend void unify_births(FiltrationValue& f1, const FiltrationValue& f2);
```

Comment:
The other function name starts with `push_to_`, probably to indicate that it modifies its first argument, while this one doesn't. The 2 functions seem similar enough that more consistent names might help? (this is more a "to be discussed", not "I want this name")

Reply:
They are very similar in 1D, but in the k-critical case, I find their role quite different.
But it is true that this one does not indicate as much that the first element is modified.

Reply:
In the k-critical case (i.e. the live region is a union of upper-right quadrants), one function performs a union and one an intersection, that feels rather similar to me.

Reply:
Ah, I never saw that as an intersection. But now that you say it, it is! So, it should be something like `intersect_births`? That sounds strange. We need perhaps something more precise. Like `unify/intersect_quadrants_by_birth` but with something better than "quadrants". Cones?

Reply:
Or just `unify` and `intersect` and let the explanation to the doc ? I don't find a good name.